libraries/AP_HAL_ChibiOS/sdcard.cpp

```diff
@@ -124,8 +125,9 @@ bool sdcard_init()
         printf("Successfully mounted SDCard (slowdown=%u)\n", (unsigned)sd_slowdown);
 
         // Create APM Directory if needed
-        AP::FS().mkdir("/APM");
-        return true;
+        if (AP::FS().mkdir("/APM") == 0) {
```

Comment:
This doesn't work, as it doesn't handle the case where the directory already exists, you would need to accept that case.